app/models/user.rb

```diff
@@ -1,7 +1,15 @@
 class User < ActiveRecord::Base
   devise :database_authenticatable, :recoverable, :rememberable, :trackable, :validatable
 
-  has_many :permissions
-  has_many :roles, through: :permissions
+  has_many :permissions do
+    def for_application(application)
+      where(application: application)
+    end
+  end
```

Comment:
Any reason this has to be a association extension and not a class method on `Permission` ?